lib/mongo/client.rb

```diff
@@ -1004,10 +1004,11 @@ def start_session(options = {})
     # @param [ Array<Hash> ] pipeline Optional additional filter operators.
     # @param [ Hash ] options The change stream options.
     #
-    # @option options [ String ] :full_document Allowed values: 'default', 'updateLookup'.
-    #   Defaults to 'default'. When set to 'updateLookup', the change notification for partial
-    #   updates will include both a delta describing the changes to the document, as well as a copy
-    #   of the entire document that was changed from some time after the change occurred.
+    # @option options [ String ] :full_document Allowed values: 'updateLookup'.
+    #   When set to 'updateLookup', the change notification for partial updates
+    #   will include both a delta describing the changes to the document, as
+    #   well as a copy of the entire document that was changed from some time
+    #   after the change occurred. The default is to not send a value.
```

Comment:
Looking at https://jira.mongodb.org/browse/DRIVERS-2231, removal of an allowed value is a (breaking) API change. If I am reading the drivers ticket correctly, the string "default" is allowed and does the same thing as no value. Therefore, a better phrasing here would be something like

```
Allowed values: nil, 'default' (behaves same as nil), 'updateLookup'.
```

... to state that the string 'default' is allowed but it behaves the same as specifying nothing.

Is there test coverage in Ruby driver for specifying both `nil` and `'default'` to this method?

Reply:
If you look at the spec change here: https://github.com/mongodb/specifications/commit/31edce69d0fed2151a08c66204b7b60afcd6ab9b
The old doc string there is equivalent to the old doc string here. (It was a copy-paste job). I just updated the old doc strings here to the new doc strings in the spec change.

Reply:
There is test coverage in `spec/mongo/collection/view/change_stream_spec.rb:97-127`. I can change that doc string

Reply:
Driver specs indicate the behavior that drivers should have going forward. They frequently do not address backwards compatibility concerns. In cases when the backwards compatibility concern is significant, it may be mentioned explicitly, for example in https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#initial-topologytype:

> If the directConnection option is not specified, **newly developed** drivers MUST behave as if it was specified with the false value.

That sentence specifically allows existing drivers to have different behavior for backwards compatibility reasons.

For this PR, the driver should specify that no value for the option is accepted. In the case of our driver we cannot remove the allowance for specifying the string `'default'` as it is already part of public API and users may be using it. We need to continue accepting this string, either passing it as is or changing it to no value (which needs to be done carefully).

Reply:
I have updated the doc string